android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/DDMFormView.kt

```diff
@@ -247,13 +247,17 @@ class DDMFormView @JvmOverloads constructor(
 	}
 
 	override fun subscribeToValueChanged(observable: Observable<Field<*>>) {
-		val autoSaveSubscription = observable.subscribeOnChange(2) { field ->
-			thing?.let {
-				presenter.onFieldValueChanged(it, formInstance, field)
-			}
-		}
-
-		subscription.add(autoSaveSubscription)
+        subscription = observable.doOnNext {
+            presenter.onFieldValueChanged(it)
+        }.debounce(2, TimeUnit.SECONDS)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ field ->
+                thing?.let {
```

Comment:
An exception should be thrown if thing is null. You could do something similar as the following code:
```
val thing = thing ?: throw Exception("No thing found")
```

Reply:
Thanks Paulo. I added the treatment like was mentioned.